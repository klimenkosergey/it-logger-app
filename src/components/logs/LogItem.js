import React from 'react';
import PropTypes from 'prop-types';
import Moment from 'react-moment';

const LogItem = ({ log, deleteLog, setCurrent }) => (
  <li className='collection-item'>
    <a
      href='#edit-log-modal'
      className={`modal-trigger ${log.attention ? 'red-text' : 'blue-text'}`}
      onClick={e => setCurrent(log)}
    >
      {log.message}
    </a>
    <br />
    <span className='grey-text'>
      <span className='black-text'>ID #{log.id}</span> last edited by{' '}
      <span className='black-text'>{log.tech}</span> on{' '}
      <Moment format={'MMMM Do YYYY, hh:mm:ss a'}>{log.date}</Moment>
    </span>
    <a
      href='#!'
      onClick={e => {
        e.preventDefault();
        deleteLog(log.id);
      }}
      className='secondary-content'
    >
      <i className='material-icons grey-text'>delete</i>
    </a>
  </li>
);

LogItem.propTypes = {
  log: PropTypes.object.isRequired,
  deleteLog: PropTypes.func.isRequired
};

export default LogItem;
