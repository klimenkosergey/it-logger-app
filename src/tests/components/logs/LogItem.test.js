import React from 'react';
import { shallow } from 'enzyme';

import LogItem from '../../../components/logs/LogItem';

let wrapper, log, deleteLogSpy, setCurrentSpy;

beforeEach(() => {
  log = {
    id: 1,
    message: 'Fixed hard drive',
    tech: 'Tony Stark',
    date: '2019-08-15T10:27:16.427Z'
  };
  deleteLogSpy = jest.fn();
  setCurrentSpy = jest.fn();
  wrapper = shallow(
    <LogItem log={log} deleteLog={deleteLogSpy} setCurrent={setCurrentSpy} />
  );
});

it('should render LogItem correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

it('should handle deleteLog', () => {
  wrapper
    .find('.secondary-content')
    .simulate('click', { preventDefault: jest.fn() });
  expect(deleteLogSpy).toHaveBeenCalledWith(log.id);
});

it('should handle setCurrent', () => {
  wrapper.find('.modal-trigger').simulate('click');
  expect(setCurrentSpy).toHaveBeenCalledWith(log);
});
