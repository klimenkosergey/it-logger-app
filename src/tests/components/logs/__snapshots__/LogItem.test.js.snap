// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LogItem correctly 1`] = `
<li
  className="collection-item"
>
  <a
    className="modal-trigger blue-text"
    href="#edit-log-modal"
    onClick={[Function]}
  >
    Fixed hard drive
  </a>
  <br />
  <span
    className="grey-text"
  >
    <span
      className="black-text"
    >
      ID #
      1
    </span>
     last edited by
     
    <span
      className="black-text"
    >
      Tony Stark
    </span>
     on
     
    <t
      ago={false}
      calendar={false}
      decimal={false}
      element={null}
      filter={[Function]}
      format="MMMM Do YYYY, hh:mm:ss a"
      fromNow={false}
      interval={60000}
      local={false}
      onChange={[Function]}
      titleFormat=""
      toNow={false}
      unit={null}
      unix={false}
      utc={false}
      withTitle={false}
    >
      2019-08-15T10:27:16.427Z
    </t>
  </span>
  <a
    className="secondary-content"
    href="#!"
    onClick={[Function]}
  >
    <i
      className="material-icons grey-text"
    >
      delete
    </i>
  </a>
</li>
`;
